packages/passport-interface/test/ApplyActions.spec.ts

```diff
@@ -0,0 +1,67 @@
+import {
+  AppendToFolderAction,
+  AppendToFolderPermission,
+  PCDActionType,
+  PCDCollection,
+  PCDPermissionType,
+  ReplaceInFolderAction
+} from "@pcd/pcd-collection";
+import { PCDPackage } from "@pcd/pcd-types";
+import { SemaphoreIdentityPCDPackage } from "@pcd/semaphore-identity-pcd";
+import { expect } from "chai";
+import { v4 as uuid } from "uuid";
+import { SubscriptionActions, applyActions } from "../src";
+
+describe("apply actions from subscription", async function () {
+  const packages: PCDPackage[] = [SemaphoreIdentityPCDPackage];
+
+  it("should record errors", async function () {
+    const pcds = new PCDCollection(packages);
+    const subscriptionActions: SubscriptionActions[] = [];
+    const data = {
+      actions: [
+        {
+          type: PCDActionType.ReplaceInFolder,
+          folder: "test",
+          pcds: []
+        } as ReplaceInFolderAction,
+        {
+          type: PCDActionType.AppendToFolder,
+          folder: "other-test",
+          pcds: []
+        } as AppendToFolderAction
+      ],
+      subscription: {
+        id: uuid(),
+        providerUrl: "http://example.com",
+        credential: undefined,
+        subscribedTimestamp: Date.now(),
+        feed: {
+          id: "1",
+          name: "Test",
+          description: "Test feed",
+          permissions: [
+            {
+              type: PCDPermissionType.AppendToFolder,
+              folder: "test"
+            } as AppendToFolderPermission
+          ]
+        }
+      }
+    };
+
+    subscriptionActions.push(data);
+
+    // We get back a map of provider/feed IDs => error counts
+    const errors = await applyActions(pcds, subscriptionActions);
+    // There's only one subscription
+    expect(errors.size).to.eq(1);
+    // The key in the error map should match the subscription ID
+    expect(Array.from(errors.keys())[0]).to.eq(data.subscription.id);
+    // But a total of two errors for this subscription, due to two
+    // actions which fail permission checks.
+    expect(Array.from(errors.values())).to.deep.eq([
+      [{ type: "permission-error" }, { type: "permission-error" }]
```

Comment:
I am assuming that the UI components know about things like permission types and error types, and will display the appropriate English text, rather than having a non-UI component specify English text. I'm going to have to revisit this in order to better handle changing permissions on subscribed feeds though.